module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -614,6 +615,9 @@ protected function getHoldingDetailsForItem($holding): array
             return $statement;
         };
         $id = $holding->id;
+        $locationData = $this->getLocationData($holding->effectiveLocationId);
```

Comment:
This lookup is already being done in formatHoldingItem(). I wonder if it would be better to construct a fake item and call that support method instead of building a complete fake array below. (I realize that some refactoring would be required, perhaps including the addition of an optional parameter for complete availability status objects... but I think that would lead to less redundant code and would reduce the odds of things getting out of sync over time). At the very least, if you decide not to refactor, formatHoldingItem() should be revised to use the new location name/code properties instead of a redundant lookup, and your new virtual holding array below should include the code as well as the name for consistency.

Reply:
I struggled with this part of the pull request—the holdings record details themselves are already a kind of fake item in that they are passed to the rest of VuFind as part of any items attached to a bib record. This is the only part where there is overlap between the holdings record fields and the item record fields. Maybe put a comment in each place cross-referencing the two places where the location UUID is translated? I think that would be clearer than trying to force a DRY refactoring here.

I'm not sure about your last comment...the code here is nearly identical to the four lines in `formatHoldingItem()` except that I didn't use a `$locationId` intermediate variable:
```php
        $locationId = $item->effectiveLocation->id;
        $locationData = $this->getLocationData($locationId);
        $locationName = $locationData['name'];
        $locationCode = $locationData['code'];
```

Reply:
Regarding the last part of my comment, I'm referring to the fact that the new 'holdingLocationName' and 'holdingLocationCode' entries in the holding details array could be used in formatHoldingItem now instead of re-fetching the location data.

Reply:
I've stared at this code a bunch. I see the somewhat duplicate code in creating the fake item, but I can't work out a refactoring path that eliminates it.  I'll gladly accept some pointers if you have any to offer.

I'll also accept pointers on updating the tests so that they pass...not sure how the changes I made cause the tests to output`/locations` rather than `/inventory/items-by-holding-id`

Reply:
I think my earlier suggestion and the test failure are related to the same thing: the code is now fetching locations twice unnecessarily, instead of reusing the locations it has already fetched. You've added a getLocationData call in line 618, but the method is then getting called a second time in line 680. Since you have access to the results from line 618 via the $holdingDetails array available to you in line 680, I think you should eliminate the second call and reuse the existing stored data. That might help!

Reply:
Oh! I get it now...`getLocationData()` isn't being called twice when there are no items (which is what I was laser-focused on)...it is when there are items attached. I pulled this code out to a helper method. Waiting to see what the tests say.